server/src/main/java/org/opensearch/search/aggregations/InternalAggregation.java

```diff
@@ -160,6 +161,16 @@ public boolean isSliceLevel() {
             return this.isSliceLevel;
         }
 
+        // For slice level partial reduce we will apply shard level `shard_size` and `shard_min_doc_count` limits whereas for coordinator
+        // level partial reduce it will use top level `size` and `min_doc_count`
+        public int getRequiredSizeLocal(TermsAggregator.BucketCountThresholds bucketCountThresholds) {
```

Comment:
@jed326 this looks good, the issue I have with having 2 methods is lack of context, consider the alternative I've tried to highlight before (yes, needs one more class):

```
LocalBucketCountThresholds asLocalBucketThresholds(TermsAggregator.BucketCountThresholds bucketCountThresholds) {
 ...
}
```

It is easy to understand what method does: it converts BucketCountThresholds to their local equivalent. The methods do the job but for someone looking to understand what is `getRequiredSizeLocal` without the context - it is literally impossible.

Reply:
@reta Made this change in the latest revision, let me know what you think!